### PR TITLE
Kube/node pool upgrade parameters

### PR DESCRIPTION
**Pull request by matprig**
Hello,

This PR contains:
- nodepool missing parameters
- update autoscale parameter versus destroy
- test acceptance
best,

### CODE REVIEW

**Review by XavierDuthil — CHANGES_REQUESTED** [review]

File: ovh/resource_cloud_project_kube_nodepool_test.go [hunk 1]
```diff
@@ -88,6 +88,56 @@ resource "ovh_cloud_project_kube_nodepool" "pool" {
...
+	    ]
+	  }
+    }
+}
+`
+
+var testAccCloudProjectKubeNodePoolConfigUpdated = `
+resource "ovh_cloud_project_kube" "cluster" {
+	service_name  = "%s"
+    name          = "%s"
```
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/resource_cloud_project_kube_nodepool_test.go:118]: fixed

File: ovh/resource_cloud_project_kube_nodepool_test.go [hunk 2]
```diff
@@ -88,6 +88,56 @@ resource "ovh_cloud_project_kube_nodepool" "pool" {
...
+	region        = "%s"
+	version = "%s"
+}
+
+resource "ovh_cloud_project_kube_nodepool" "pool" {
+	service_name  = ovh_cloud_project_kube.cluster.service_name
+	kube_id       = ovh_cloud_project_kube.cluster.id
+    name          = ovh_cloud_project_kube.cluster.name
```
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/resource_cloud_project_kube_nodepool_test.go:126]: fixed

File: ovh/types_cloud_project_kube_nodepool.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+func (s *CloudProjectKubeNodePoolCreateOpts) String() string {
+	return fmt.Sprintf("%s(%s): %d/%d/%d", *s.Name, s.FlavorName, *s.DesiredNodes, *s.MinNodes, *s.MaxNodes)
+}
+
+type Taint struct {
+	//"NoExecute"
+	//"NoSchedule"
+	//"PreferNoSchedule"
```
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/types_cloud_project_kube_nodepool.go:102]: better have an enumeration, i will do that

File: ovh/types_cloud_project_kube_nodepool.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+			Labels:      map[string]string{},
+		},
+		Spec: &CloudProjectKubeNodePoolTemplateSpec{
+			Taints:        []Taint{},
+			Unschedulable: false,
+		},
+	}
+
+	templateSet := i.(*schema.Set).List()
+	for _, inter := range templateSet {
```
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/types_cloud_project_kube_nodepool.go:50]: refactored

File: ovh/types_cloud_project_kube_nodepool.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
```
[the remark this answers is not in the document]
Comment (XavierDuthil) [reply ovh/types_cloud_project_kube_nodepool.go:53]: Okay, can you remove it then please? :p

File: ovh/resource_cloud_project_kube_nodepool_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (XavierDuthil) [reply ovh/resource_cloud_project_kube_nodepool_test.go:118]: No it's even worse now :(

File: ovh/resource_cloud_project_kube_nodepool_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (XavierDuthil) [reply ovh/resource_cloud_project_kube_nodepool_test.go:126]: I don't see it better

File: ovh/types_cloud_project_kube_nodepool.go [hunk 6]
```diff
@@ -96,13 +98,54 @@ func (s *CloudProjectKubeNodePoolCreateOpts) String() string {
 	return fmt.Sprintf("%s(%s): %d/%d/%d", *s.Name, s.FlavorName, *s.DesiredNodes, *s.MinNodes, *s.MaxNodes)
 }
 
+type EffectType int
+
+const (
```
Comment (XavierDuthil) [inline ovh/types_cloud_project_kube_nodepool.go:103]: Not a fan of types and constants defined mid-file, but ok

File: ovh/types_cloud_project_kube_nodepool.go [hunk 7]
```diff
@@ -96,13 +98,54 @@ func (s *CloudProjectKubeNodePoolCreateOpts) String() string {
 	return fmt.Sprintf("%s(%s): %d/%d/%d", *s.Name, s.FlavorName, *s.DesiredNodes, *s.MinNodes, *s.MaxNodes)
 }
 
+type EffectType int
```
Comment (XavierDuthil) [inline ovh/types_cloud_project_kube_nodepool.go:101]:
```suggestion
type TaintEffectType int
```

File: ovh/types_cloud_project_kube_nodepool.go [hunk 8]
```diff
@@ -96,13 +98,54 @@ func (s *CloudProjectKubeNodePoolCreateOpts) String() string {
...
+	return toString[e]
+}
+
+var toString = map[EffectType]string{
+	NoExecute:        "NoExecute",
+	NoSchedule:       "NoSchedule",
+	PreferNoSchedule: "PreferNoSchedule",
+}
+
+var EffecTypetoID = map[string]EffectType{
```
Comment (XavierDuthil) [inline ovh/types_cloud_project_kube_nodepool.go:119]:
```suggestion
var EffecTypeToID = map[string]EffectType{
```

**Review by XavierDuthil — CHANGES_REQUESTED** [review]

File: ovh/resource_cloud_project_kube_nodepool_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/resource_cloud_project_kube_nodepool_test.go]: there was a mess with tab and space characters...

refixed

File: ovh/types_cloud_project_kube_nodepool.go (hunk 6 above)
Comment (matprig) [reply ovh/types_cloud_project_kube_nodepool.go:103]: fixed tab & space chars
and reorg

File: ovh/types_cloud_project_kube_nodepool.go (hunk 7 above)
Comment (matprig) [reply ovh/types_cloud_project_kube_nodepool.go:101]: yes and renamed also EffecTypetoID => TaintEffecTypetoID

File: ovh/resource_cloud_project_kube_nodepool_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/resource_cloud_project_kube_nodepool_test.go]: refixed

File: ovh/types_cloud_project_kube_nodepool.go (hunk 8 above)
Comment (matprig) [reply ovh/types_cloud_project_kube_nodepool.go]: yes

File: ovh/types_cloud.go [hunk 9]
```diff
@@ -117,7 +117,7 @@ func (u CloudProjectUser) ToMap() map[string]interface{} {
 	obj := make(map[string]interface{})
 	obj["creation_date"] = u.CreationDate
 	obj["description"] = u.Description
-	//Dont set password as it must be set only at creation time
+	// Dont set password as it must be set only at creation time
```
Comment (scraly) [inline ovh/types_cloud.go:120]:
```suggestion
	// Don't set password as it must be set only at creation time
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Comment by scraly** [comment]
```
$ make testacc TESTARGS="-run TestAccCloudProjectKubeNodePool_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run TestAccCloudProjectKubeNodePool_basic -timeout 240m
?   	github.com/ovh/terraform-provider-ovh	[no test files]
=== RUN   TestAccCloudProjectKubeNodePool_basic
--- PASS: TestAccCloudProjectKubeNodePool_basic (562.85s)
    provider_test.go:218: Read Cloud Project /cloud/project/a212a1e43b614c4ba27a247b890fcf59 -> status: 'ok', desc: 'xxx_xxxxx_xxx'
PASS
ok  	github.com/ovh/terraform-provider-ovh/ovh	564.035s
?   	github.com/ovh/terraform-provider-ovh/ovh/helpers	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/ovh/terraform-provider-ovh/ovh/helpers/hashcode	0.271s [no tests to run]
```

```
$ make testacc TESTARGS="-run TestAccCloudProjectKubeEmptyVersion_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run TestAccCloudProjectKubeEmptyVersion_basic -timeout 240m
?   	github.com/ovh/terraform-provider-ovh	[no test files]
=== RUN   TestAccCloudProjectKubeEmptyVersion_basic
--- PASS: TestAccCloudProjectKubeEmptyVersion_basic (293.14s)
    provider_test.go:218: Read Cloud Project /cloud/project/a212a1e43b614c4ba27a247b890fcf59 -> status: 'ok', desc: 'xxxx_xxx_xxx'
PASS
ok  	github.com/ovh/terraform-provider-ovh/ovh	294.327s
?   	github.com/ovh/terraform-provider-ovh/ovh/helpers	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/ovh/terraform-provider-ovh/ovh/helpers/hashcode	0.447s [no tests to run]
```

✔🙂

**Review by scraly — APPROVED** [review]
After several tests, it's OK for me :)

**Pull request merged**